src/octoprint/util/gcodeInterpreter.py

```diff
@@ -122,13 +130,15 @@ def _load(self, gcodeFile, printer_profile, throttle=None):
 					z = getCodeFloat(line, 'Z')
 					e = getCodeFloat(line, 'E')
 					f = getCodeFloat(line, 'F')
-					#Check if x,y or z exceeds printing area
-					width=float(printer_profile["volume"]["width"])
-					depth=float(printer_profile["volume"]["depth"])
-					height=float(printer_profile["volume"]["height"])
-					if not self.warn and (x>width or y>depth or z>height):
-						self.warn=True 
-						self._logger.warn("Object is bigger than the printing area")
+					#Add value to list in case move includes extrusion
+					if e is not None and (e-previousE)>0:
+						if x is not None:
+							xList.append(x)
```

Comment:
Using lists for this is very risky. You are basically creating a full representation of the model in memory, which depending on the size of the model can be too much for the hardware you are running on (Pi1). At the very least you should be using a `set` here, which makes sure to not keep duplicates, but even that has the risk of causing memory issues.

You are only interested in the min and max values here, so just track those on the go:

``` python
min_x = None
min_y = None
min_z = None
max_x = None
max_y = None
max_z = None

# ...

if e is not None and (e - previousE) > 0:
    min_x = x if min_x is None or x < min_x else min_x
    max_x = x if max_x is None or x > max_x else max_x
    min_y = y if min_y is None or y < min_y else min_y
    max_y = y if max_y is None or y > max_y else max_y
    previousE = e

if z is not None:
    min_z = z if min_z is None or z < min_z else min_z
    max_z = z if max_z is None or z > max_z else max_z

# ...

self.minX = min_x
self.maxX = max_x
#...
```